### PR TITLE
Support OVStableDiffusionPipelineBase to Load Textual Inversion in Runtime

### PR DESCRIPTION
**Pull request by sammysun0711**
This PR aims to support OVStableDiffusionPipelineBase to load pretrained textual inversion embeddings in runtime to save disk storage while keep flexibility:
 
- `Save disk storage`: We only need to save a SD baseline model exported to OV IR (e.g.: SD-1.5 ~5GB) and textual embedding (~10KB-100KB), instead of multiple SD with textual inversion embeddings merged (~n *5GB), since disk storage is very limited especially for edge/client use case.
- `Flexibility`: We can load (multiple) pre-trained textual inversion embeddings in SD baseline model in runtime quickly, which supports combination of embeddings and avoid to mess up baseline model.

So I implemented OVTextualInversionLoaderMixin based on [diffusers.loaders.TextualInversionLoaderMixin](https://github.com/huggingface/diffusers/blob/29ece0db7966a8d25fdd8f314a13096194bc5568/src/diffusers/loaders.py#L577) as follow: 
- Support textual embeddings saved as `*.bin`, `*.pt`, `*.safetensors`
- Update tokenizer with new token id to expand vocab size
- Update text encoder embeddings via `InvertTextEmbedding` based on OV ngraph transformation [OpenVINO ngraph transformation](https://docs.openvino.ai/2023.0/openvino_docs_Extensibility_UG_matcher_pass.html)
- Verified with SD-1.5, SD-2.1, SDXL-base-1.0 pipeline

The workflow of `InvertTextEmbedding` can be summary as follow:
insert_text_embedding_ngraph_transformation

We can visualized 3 phase of `InvertTextEmbedding` in SD-1.5 with single textual inversion embedding <cat-toy> with shape (,768) in graph with Netron: 

visualized_text_embedding_in_netron

### CODE REVIEW

**Comment by sammysun0711** [comment]
Here is some text inversion examples:
1. [SD 1.5](https://huggingface.co/runwayml/stable-diffusion-v1-5) + [sd-concepts-library/cat-toy](https://huggingface.co/sd-concepts-library/cat-toy)
```
model_id = "runwayml/stable-diffusion-v1-5"
prompt = "A <cat-toy> back-pack"
np.random.seed(42)

# Run pipeline without textual inversion
pipe = OVStableDiffusionPipeline.from_pretrained(model_id, compile=False, cache_dir=None)
pipe.compile()
image1 = pipe(prompt, num_inference_steps=50).images[0]
image1.save("sd_v1.5_without_cat_toy_ti.png")

# Run pipeline with textual inversion
pipe.clear_requests()
pipe.load_textual_inversion("sd_concepts/cat-toy", "<cat-toy>")
pipe.compile()
image2 = pipe(prompt, num_inference_steps=50).images[0]
image2.save("sd_v1.5_with_cat_toy_ti.png")
```
SD 1.5 baseline
sd_v1 5_without_ti
SD 1.5 baseline + cat-toy TI
sd_v1 5_with_ti

2. [SD 1.5](https://huggingface.co/runwayml/stable-diffusion-v1-5) + [sd-concepts-library/cat-toy](https://huggingface.co/sd-concepts-library/cat-toy) + [charturnerv2](https://civitai.com/models/3036/charturner-character-turnaround-helper) + [EasyNegative](https://huggingface.co/datasets/gsdf/EasyNegative)
```
model_id = "runwayml/stable-diffusion-v1-5"
prompt = "charturnerv2, multiple views of the same character in the same outfit, a character turnaround of a beautiful woman wearing a black jacket and red shirt, best quality, intricate details."
np.random.seed(42)

# Run pipeline without charturnerv2 textual inversion
pipe = OVStableDiffusionPipeline.from_pretrained(model_id, compile=False, cache_dir=None)
pipe.compile()
image3 = pipe(prompt, num_inference_steps=50).images[0]
image3.save("sd_v1.5_without_charturner_ti.png")

# Run pipeline with charturner textual inversion
pipe.clear_requests()
pipe.load_textual_inversion(["./charturnerv2.pt"], ["charturnerv2"])
pipe.compile()
image4 = pipe(prompt, num_inference_steps=50).images[0]
image4.save("sd_v1.5_with_charturner_ti.png")

# Run pipeline with charturner + EasyNegative textual inversion 
pipe.clear_requests()
pipe.load_textual_inversion(["./charturnerv2.pt", "./EasyNegative/EasyNegative.pt"], ["charturnerv2", "EasyNegative"])
pipe.compile()
negative_prompt = "EasyNegative"
image5 = pipe(prompt, num_inference_steps=50, negative_prompt=negative_prompt).images[0]
image5.save("sd_v1.5_with_charturner_easynegative_ti.png")

```
SD 1.5 baseline
sd_v1 5_without_ti_new

SD 1.5 baseline + charturnerv2 TI 
sd_v1 5_with_charturner_ti

SD 1.5 baseline + charturnerv2 TI + EasyNegative TI
sd_v1 5_with_charturner_easy_engative_ti

**Comment by sammysun0711** [comment]
3. [SD 2.1](https://huggingface.co/stabilityai/stable-diffusion-2-1) + [midjourney_sd_2_0](minimaxir/midjourney_sd_2_0)
```
from optimum.intel import OVStableDiffusionPipeline
import numpy as np

model_id = "stabilityai/stable-diffusion-2-1"
prompt = "A <midjourney> style photo of an astronaut riding a horse on mars"
np.random.seed(42)

# Run pipeline without midjourney textual inversion
pipe = OVStableDiffusionPipeline.from_pretrained(model_id, compile=False, cache_dir=None)
pipe.compile()
image1 = pipe(prompt, num_inference_steps=50).images[0]
image1.save("sd_v2.1_without_midjourney_ti.png")

# Run pipeline with midjourney textual inversion
pipe.clear_requests()
pipe.load_textual_inversion("midjourney_sd_2_0", "<midjourney>")
pipe.compile()
image2 = pipe(prompt, num_inference_steps=50).images[0]
image2.save("sd_v2.1_with_midjourney_ti.png")
```
SD 2.1 baseline

sd_v2 1_without_ti

SD 2.1 + Midjourney TI
sd_v2 1_with_ti

**Comment by sammysun0711** [comment]
4. [SDXL-1.0-base](https://huggingface.co/stabilityai/stable-diffusion-xl-base-1.0) + [charturnerv2](https://civitai.com/models/3036/charturner-character-turnaround-helper)
```
from optimum.intel import OVStableDiffusionXLPipeline
import numpy as np

model_id = "stabilityai/stable-diffusion-xl-base-1.0"
prompt = "charturnerv2, multiple views of the same character in the same outfit, a character turnaround of a beautiful woman wearing a black jacket and red shirt, best quality, intricate details."
np.random.seed(0)

pipe = OVStableDiffusionXLPipeline.from_pretrained(model_id, export=False, compile=False, cache_dir=None)

# Run pipeline without textual inversion
pipe.compile()
image1 = pipe(prompt, num_inference_steps=50).images[0]
image1.save("sdxl_base_1.0_without_charturnerv2_ti.png")

# Run pipeline with textual inversion
pipe.clear_requests()
pipe.load_textual_inversion("./charturnerv2.pt", "charturnerv2")
pipe.compile()
image2 = pipe(prompt, num_inference_steps=50).images[0]
image2.save("sdxl_base_1.0_with_charturnerv2_ti.png")

SDXL-1.0-Base baseline:
![sdxl_base_1 0_with_ti](https://github.com/huggingface/optimum-intel/assets/22404054/c1131dd3-f059-4670-99f7-de3201534aa0)

SDXL-1.0-Base + Charturnerv2
![sdxl_base_1 0_without_ti](https://github.com/huggingface/optimum-intel/assets/22404054/69adb5d7-2381-4778-8bd7-bd5543fb71e1)

**Comment by sammysun0711** [comment]
@echarlaix, could you please kindly review it?

cc @AlexKoff88, @helena-intel

**Comment by AlexKoff88** [comment]
@sammysun0711, please fixe the code quality by running `make style`

**Comment by sammysun0711** [comment]
> @sammysun0711, please fixe the code quality by running `make style`

Thanks for your tips. I've fixed format with `make style` and update PR.

**Review by AlexKoff88 — APPROVED** [review]

**Review by fxmarty — APPROVED** [review]
@sammysun0711 This is a great addition! Do you think we could add the snippets you provided to the documentation? For example adding a section here　https://huggingface.co/docs/optimum/main/en/intel/inference#imagetoimage ?

**Comment by sammysun0711** [comment]
@fxmarty thanks for your advice.  I added textual inversion snippets to documentation. 
BTW, I create a PR to add images for textual inversion docs: https://huggingface.co/datasets/optimum/documentation-images/discussions/1
Could you please kindly review it, Thanks!

**Comment by fxmarty** [comment]
Failing tests are unrelated, thank you!
